mscore/uploadscoredialog.cpp

```diff
@@ -95,8 +95,8 @@ void UploadScoreDialog::upload(int nid)
      {
      Score* score = mscore->currentScore()->masterScore();
      const QString scoreTitle = title->text().trimmed().isEmpty() ? score->title() : title->text();
-     QString path = QDir::tempPath() + QString("/temp_%1.mscz").arg(qrand() % 100000);
-     if(mscore->saveAs(score, true, path, "mscz")) {
+     QString path = QDir::tempPath() + "/" + mscore->currentScore()->masterScore()->fileInfo()->fileName();
```

Comment:
Maybe it would be more safe to add a check that this file doesn't exist?

Reply:
Why? It is still a temp file in a temp dir.

Reply:
This file could potentially be created by another program or by user explicitly. For example, this file could be opened from a web browser which would save it somewhere to a temporary location, and later user could decide to try uploading it. It is probably not a very likely scenario but still it is better to be accurate here.

Reply:
/tmp is not a safe place to store anything of value.
And what to do if it exists? Never check for error conditions you can't handle ;-)
Append a random number maybe? Could still exist but reduces the likelyhood even further

Reply:
Yes, maybe append a number if file exists. Or we can create a directory in temporary location with random name instead and store a file with normal name within it.

Reply:
prepended a random number to the filename now, if it exists without that